skimage/feature/texture.py

```diff
@@ -97,11 +99,21 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
+    if levels is None:
+        # if levels is not given, we assume that there are [0, image_max -1] levels to 
+        # be considered. There would be only zeros for all other rows and columns. 
```

Comment:
Please, refactor the comments here and below: make them impersonal, fix syntax, etc.